client/me/social-login/index.jsx

```diff
@@ -94,18 +104,40 @@ class SocialLogin extends Component {
 						{ errorUpdatingSocialConnection.message }
 					</Notice>
 				) }
+
 				<CompactCard>
 					{ translate(
-						'You’ll be able to log in faster by linking your WordPress.com account with your ' +
-							'social networks. We’ll never post without your permission.'
+						'You’ll be able to log in faster by linking your WordPress.com account with the following ' +
+							'third-party services. We’ll never post without your permission.'
 					) }
 				</CompactCard>
+
 				{ this.renderGoogleConnection() }
+
+				{ config.isEnabled( 'sign-in-with-apple' ) && this.renderAppleConnection() }
 			</div>
 		);
 	}
 
-	renderActionButton( onClickAction = null ) {
+	renderAppleActionButton( onClickAction = null ) {
+		const { isUserConnectedToApple, isUpdatingSocialConnection, translate } = this.props;
+		const buttonLabel = isUserConnectedToApple ? translate( 'Disconnect' ) : translate( 'Connect' );
+		const disableButton = isUpdatingSocialConnection || this.state.fetchingUser;
+
+		return (
+			<FormButton
+				className="social-login__button button"
+				disabled={ disableButton }
+				compact={ true }
+				isPrimary={ ! isUserConnectedToApple }
+				onClick={ onClickAction }
+			>
+				{ buttonLabel }
+			</FormButton>
+		);
+	}
```

Comment:
We're introducing lots of duplicate code here whereas the only difference between `renderAppleActionButton()` and `renderGoogleActionButton()` lies in `isUserConnectedToApple`. I'm sure we can improve that 😉.